src/Symfony/Component/HttpKernel/DataCollector/DumpDataCollector.php

```diff
@@ -55,6 +59,8 @@ public function __construct(Stopwatch $stopwatch = null, $fileLinkFormat = null,
             &$this->isCollected,
             &$this->clonesCount,
         );
+
+        $this->sourceContextProvider = new SourceContextProvider($this->charset);
```

Comment:
this defeats dependency injection, that's why I proposed to get it through `$serverDumper`